runtime/interpreter/encoding_test.go

```diff
@@ -2895,6 +3288,49 @@ func TestEncodeDecodeSomeValue(t *testing.T) {
 			},
 		)
 	})
+
+	t.Run("SomeValue{SomeValue{large_array_stored_separately}}", func(t *testing.T) {
+		t.Parallel()
+
+		inter := newTestInterpreter(t)
+
+		maxInlineElementSize := atree.MaxInlineArrayElementSize()
+
+		expectedArray :=
+			NewArrayValue(
+				inter,
+				EmptyLocationRange,
+				ConstantSizedStaticType{
+					Type: PrimitiveStaticTypeAnyStruct,
+					Size: 0,
+				},
+				testOwner,
+				NewUnmeteredStringValue(strings.Repeat("a", int(maxInlineElementSize/2))),
+				NewUnmeteredStringValue(strings.Repeat("b", int(maxInlineElementSize/2))),
+			)
+
+		expected := NewUnmeteredSomeValueNonCopying(NewUnmeteredSomeValueNonCopying(expectedArray))
+
+		testEncodeDecode(t,
+			encodeDecodeTest{
+				storage:              inter.Storage(),
+				value:                expected,
+				maxInlineElementSize: maxInlineElementSize,
+				encoded: []byte{
+					// tag
+					0xd8, CBORTagSomeValueWithNestedLevels,
+					// array of 2 elements
+					0x82,
+					// nested levels: 2
+					0x02,
+					// value
+					0xd8, atree.CBORTagSlabID,
+					// storage ID
+					0x50, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x42, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1,
+				},
+			},
+		)
+	})
```

Comment:
Can you please also add a test for the case where there are multiple "parts" of nest `Some`s, e.g. `Some(Some([Some(Some(Some(1)))]))` (2 Some, Array, 3 Some), to test that `getInnermostStorable` does not traverse returns the array and not the integer.